mm/mm_heap/mm_sem.c

```diff
@@ -80,9 +80,21 @@ bool mm_takesemaphore(FAR struct mm_heap_s *heap)
 
   if (up_interrupt_context())
     {
-      /* Can't take semaphore in the interrupt handler */
+#if !defined(CONFIG_SMP) && defined(CONFIG_DEBUG_MM)
```

Comment:
Let's remove define(CONFIG_DEBUG_MM), it isn't good to change the important action with/without CONFIG_DEBUG_MM

Reply:
I was under the impression the heap check and need for this code is a DBUG only feature. Is that still correct?

Reply:
Yes, taking semaphore in the interrupt context should happen only when we enable memory debugging.

Reply:
So why remove CONFIG_DEBUG_MM?